service/frontend/nexus_handler_test.go

```diff
@@ -221,3 +249,54 @@ func TestNexusInterceptRequeset_GlobalRateLimited_ResultsInResourceExhausted(t *
 	require.Equal(t, 1, len(snap["test"]))
 	require.Equal(t, map[string]string{"outcome": "global_rate_limited"}, snap["test"][0].Tags)
 }
+
+func TestNexusInterceptRequest_ForwardingDisabled_ResultsInBadRequest(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	var err error
+	oc := newOperationContext(contextOptions{
+		namespaceState:          enumspb.NAMESPACE_STATE_REGISTERED,
+		namespacePassive:        true,
+		quota:                   1,
+		namespaceRateLimitAllow: true,
+		rateLimitAllow:          true,
+		redirectAllow:           false,
+	})
+	err = oc.interceptRequest(ctx, &matchingservice.DispatchNexusTaskRequest{}, nexus.Header{})
+	var handlerError *nexus.HandlerError
+	require.ErrorAs(t, err, &handlerError)
+	require.Equal(t, nexus.HandlerErrorTypeBadRequest, handlerError.Type)
```

Comment:
It's a bad request because the namespace is passive and request forwarding is disabled for this namespace, so we cannot handle this request even on retry. `serviceerror.NamespaceNotActive` is a `FailedPrecondition` error, so I thought it was the user's responsibility to make sure their request is going to the right place.

Reply:
Yeah, that makes sense to some degree.
For Temporal Cloud this should be an error that counts against our SLA, it means that something is misconfigured.
I'm on the fence about the status code.